app/ts/PreferencesTab/PreferencesUtil.ts

```diff
@@ -67,3 +68,23 @@ export function buildScenarioWithPreferences(
   };
   return {..._.omit(scenario, ['state']), state: newState};
 }
+
+export function filterScenariosWithPvfs(
+  scenarios: Record<string, IMcdaScenario>,
+  problemCriteria: Record<string, IProblemCriterion>
+): Record<string, IMcdaScenario> {
+  return _.pickBy(scenarios, (scenario: IMcdaScenario) =>
+    _.every(
+      scenario.state.problem.criteria,
+      (criterion: IScenarioCriterion, criterionId: string) => {
+        const scenarioPvf = getScenarioPvf(criterionId, scenario);
+        const problemPvf = getProblemPvf(criterionId, problemCriteria);
```

Comment:
We should really start splitting out these things on upload :'( All the special casing :(